pulumiyaml/program.go

```diff
@@ -0,0 +1,118 @@
+package pulumiyaml
+
+import "fmt"
+
+type Program struct {
+	// Defaults to "test"
+	Name string `yaml:"name,omitempty"`
+	// Defaults to "yaml"
+	Runtime     string              `yaml:"runtime,omitempty"`
+	Description string              `yaml:"description,omitempty"`
+	Config      map[string]any      `yaml:"config"`
+	Plugins     Plugins             `yaml:"plugins"`
+	Resources   map[string]Resource `yaml:"resources"`
+}
+
+func NewProgram() *Program {
+	return &Program{
+		Name:      "test",
+		Runtime:   "yaml",
+		Config:    make(map[string]any),
+		Resources: make(map[string]Resource),
+	}
+}
+
+// AddResource adds a resource to the program by name.
```

Comment:
I'm skeptical that we need these methods. They don't provide that much use beyond `p.Resources`.

If we were to have them, I think they would be more useful as a fluent style:

```go
func (p *Program) AddResource(t *testing.T, name string, resource Resource) *Program {
```

That way people can write:

```go
program := pulumiyaml.NewProgram().
	AddResource(t, "res1", res1).
	AddResource(t, "res2", res2)
```

Reply:
I think we'll move to use the structs from the pulumi-yaml repo, so these will go away. Can always add helpers later I guess once we've got some concrete use cases.